code/game/atoms.dm

```diff
@@ -293,6 +301,69 @@ its easier to just keep the beam vertical.
 /atom/proc/fire_act()
 	return
 
+/atom/proc/scp914_act(var/mode) //DON'T OVERRIDE, override scp914_act_on_type instead
+	var/list/chosen_mode = modes_914[mode]
+	if(chosen_mode && chosen_mode.len > 0)
+		var/chosen = pick(chosen_mode)
+		var/atom/chosen_obj = new chosen(loc)
+		if(chosen_obj)
+			return scp914_act_on_type(chosen_obj)
+	else
+		switch(mode)
+			if(1) //TODO: 1 and 2 should be refactored to fit MATERIAL datums once they are ready
+				var/atom/choice = new parent_type
+				if(choice)
+					var/atom/second_choice = choice.parent_type
+					if(second_choice)
+						var/atom/atombychoice = new second_choice(loc)
+						qdel(choice)
+						if(atombychoice)
+							return scp914_act_on_type(atombychoice)
+			if(2)
+				var/choice = parent_type
+				if(choice)
+					var/atom/atombychoice = new choice(loc)
+					if(atombychoice)
+						return scp914_act_on_type(atombychoice)
+			if(3)
+				var/list/choices = subtypesof(parent_type)
+				if(choices && choices.len > 0)
+					for(var/atom/typo in choices)
+						var/atom/choico = new typo
+						if("[choico.parent_type]" != "[parent_type]")
```

Comment:
DM allows you to compare type paths. Sure you need to convert to string here?

Reply:
They do accepts children, something i want to avoid.

Reply:
Is that a hint at our staff? I tried working things out with you, if you had returned my messages.